wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -197,10 +198,23 @@ def deploy_template(self, template, tags=None, password='smartvm', **kwargs):
         # create project
         # assuming this is cfme installation and generating project name
         proj_id = "".join(choice(string.digits + string.lowercase) for _ in range(6))
-        proj_name = "{t}-project-{proj_id}".format(t=template, proj_id=proj_id)
+
+        # for sprout
+        if 'vm_name' in kwargs:
+            proj_name = kwargs['vm_name']
+        else:
+            proj_name = "{t}-project-{proj_id}".format(t=template, proj_id=proj_id)
+
         proj_url = "{proj}.{base_url}".format(proj=proj_id, base_url=self.base_url)
         self.logger.info("unique id %s, project name %s", proj_id, proj_name)
+
+        progress_callback = kwargs.get('progress_callback', None)
+        if not progress_callback:
+            progress_callback = partial(self._progress_log_callback, self.logger,
```

Comment:
Consider collapsing this into the default/fallback value on `kwargs.get('progress_callback')`